### ISSUE TITLE
Raspberry Pi 4 Support

### ISSUE THREAD

**Issue by illegalprime** [illegalprime/nixos-on-arm#15]
is it in the upstream kernel?
how much work would t be to add it?
let's find out!

## Kernel

RPi4 things don't seem to be upstreamed yet, but DTS and defconfigs are available here: https://github.com/raspberrypi/linux

This /should be/ pretty easy.

## UBoot

not sure yet..

**Comment by illegalprime** [comment]
didn't know upstream was already making good progress
https://github.com/NixOS/nixpkgs/pull/68265

It might be easy for u-boot to be patched, will investigate more.

**Comment by samueldr** [comment]
U-Boot will get support after the mainline kernel has initial support, from what I remember reading. Something about waiting on the kernel for the device tree source files, as they go `kernel` → `u-boot` or something like that.

* * *

Here's the patch series on their patchwork.

 * https://patchwork.ozlabs.org/cover/1136356/

**Comment by illegalprime** [comment]
Just saw this, thanks for the info/link!

**Comment by jpraczyk** [comment]
Just an fyi @illegalprime, it seems that u-boot 2019.10 already supports rpi 4.

**Comment by samueldr** [comment]
Yes it should, I updated u-boot, and also opened a PR with the added u-boot for rpi4, but..

 * This is not tested
 * It lacks the changes to the generic image

https://github.com/NixOS/nixpkgs/pull/70796

Changes should be relatively trivial, I should be able to get back to it by November, unless someone else got it going before I have the time to work on that.
